src/components/v2/V2Project/V2ManageTokensSection/V2RedeemModal.tsx

```diff
@@ -0,0 +1,262 @@
+import { t, Trans } from '@lingui/macro'
+import { Form, Space } from 'antd'
+import { useForm } from 'antd/lib/form/Form'
+
+import InputAccessoryButton from 'components/shared/InputAccessoryButton'
+import FormattedNumberInput from 'components/shared/inputs/FormattedNumberInput'
+
+import { CSSProperties, useContext, useState } from 'react'
+import { formatPercent, formatWad, fromWad, parseWad } from 'utils/formatNumber'
+
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+import { tokenSymbolText } from 'utils/tokenSymbolText'
+import useTotalBalanceOf from 'hooks/v2/contractReader/TotalBalanceOf'
+import { NetworkContext } from 'contexts/networkContext'
+import { ThemeContext } from 'contexts/themeContext'
+import { formatRedemptionRate } from 'utils/v2/math'
+import CurrencySymbol from 'components/shared/CurrencySymbol'
+import { useETHReceivedFromTokens } from 'hooks/v2/contractReader/ETHReceivedFromTokens'
+import { V2_CURRENCY_USD } from 'utils/v2/currency'
+import { useRedeemTokensTx } from 'hooks/v2/transactor/RedeemTokensTx'
+import TransactionModal from 'components/shared/TransactionModal'
+
+const statsStyle: CSSProperties = {
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'baseline',
+}
+
+// This doubles as the 'Redeem' and 'Burn' modal depending on if project has overflow
+export default function V2RedeemModal({
+  visible,
+  onOk,
+  onCancel,
+}: {
+  visible?: boolean
+  onOk: VoidFunction | undefined
+  onCancel: VoidFunction | undefined
+}) {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+  const { userAddress } = useContext(NetworkContext)
+  const {
+    tokenSymbol,
+    fundingCycle,
+    primaryTerminalCurrentOverflow,
+    projectId,
+    totalTokenSupply,
+    distributionLimitCurrency,
+    fundingCycleMetadata,
```

Comment:
@johnnyd-eth fyi that `fundingCycleMetadata` exists in project context, so we don't need to decode metadata from `fundingCycle.metadata`